saw-remote-api/python/saw/llvm.py

```diff
@@ -458,6 +476,16 @@ def field(base : SetupVal, field_name : str) -> 'FieldVal':
     """Returns a ``FieldVal`` using the field ``field_name`` of the struct ``base``."""
     return FieldVal(base, field_name)
 
+# It's tempting to name this `global` to mirror SAWScript's `llvm_global`,
+# but that would clash with the Python keyword `global`.
+def global_lvalue(name: str) -> 'GlobalLValueVal':
+    """Returns a ``GlobalLValueVal`` representing a pointer to the named global ``name``."""
```

Comment:
The `lvalue` in `global_lvalue` is bugging me a little here. I'm looking at the [saw-script description of llvm_global](https://github.com/GaloisInc/saw-script/blob/99a0dad7a75153cfacca27f419ced91cb3086939/src/SAWScript/Interpreter.hs#L2535) and `lvalue` isn't mentioned. I _do_ see, however, internally in our implementation of the rpc server we have chosen to describe things as `lvalue`s, but interestingly saw-script proper doesn't seem to use the name anywhere. (At least I haven't found it in my perusing thus far.)

Is `lvalue` actually a better description to have users think in terms of here? Or would something slightly more general (e.g., `global_var`, `global_named_val`, `global_name`) be better?

Reply:
I don't think we gain much by using `GlobalLValueVal` as the return type here instead of the more generic `SetupVal`. Unless we have a desire to talk about computations over `GlobalLValueVal`s specifically, I think it will be clearer to use the parent class `SetupVal` for the return type, since that will be more illustrative of the kinds functions you can use the value with. (And if we change the return type, obviously the docstring should reflect that change.)

Reply:
> Is `lvalue` actually a better description to have users think in terms of here? Or would something slightly more general (e.g., `global_var`, `global_named_val`, `global_name`) be better?

I'm not wedded to the use of `lvalue` in particular. I couldn't name it "`global`", since that would clash with the Python keyword of the same name, so I picked the nearest word on hand to attach as a suffix to disambiguate it. (I'm not entirely sure why the remote API uses the word `lvalue`, to be honest.) I've renamed this to `global_var`.

> I don't think we gain much by using `GlobalLValueVal` as the return type here instead of the more generic `SetupVal`.

Fine by me. I only made it return `GlobalLValueVal` for consistency with the surrounding functions, which also return subclasses of `SetupVal`. I've made them all return `SetupVal` in a separate commit.

Reply:
> Fine by me. I only made it return `GlobalLValueVal` for consistency with the surrounding functions, which also return subclasses of `SetupVal`. I've made them all return `SetupVal` in a separate commit.

Oh, apologies, I should have looked at the other definitions and noticed that. I think it is better with your latest modifications making everything return a `SetupVal`